src/many-ledger/src/storage/ledger_tokens.rs

```diff
@@ -0,0 +1,519 @@
+use crate::error;
+use crate::migration::tokens::TOKEN_MIGRATION;
+use crate::storage::iterator::LedgerIterator;
+use crate::storage::{key_for_account_balance, LedgerStorage, SYMBOLS_ROOT};
+use many_error::ManyError;
+use many_identity::Address;
+use many_modules::events::EventInfo;
+use many_modules::ledger::extended_info::{ExtendedInfoKey, TokenExtendedInfo};
+use many_modules::ledger::{
+    TokenAddExtendedInfoArgs, TokenAddExtendedInfoReturns, TokenCreateArgs, TokenCreateReturns,
+    TokenInfoArgs, TokenInfoReturns, TokenRemoveExtendedInfoArgs, TokenRemoveExtendedInfoReturns,
+    TokenUpdateArgs, TokenUpdateReturns,
+};
+use many_types::ledger::{Symbol, TokenAmount, TokenInfo, TokenInfoSummary, TokenInfoSupply};
+use many_types::{AttributeRelatedIndex, Either, SortOrder};
+use merk::{BatchEntry, Op};
+use std::collections::{BTreeMap, BTreeSet};
+use std::str::FromStr;
+
+pub const SYMBOLS_ROOT_DASH: &str = const_format::concatcp!(SYMBOLS_ROOT, "/");
+pub const TOKEN_IDENTITY_ROOT: &str = "/config/token_identity";
+pub const TOKEN_SUBRESOURCE_COUNTER_ROOT: &str = "/config/token_subresource_id";
+
+pub fn key_for_symbol(symbol: &Symbol) -> Vec<u8> {
+    format!("/config/symbols/{symbol}").into_bytes()
+}
+
+pub fn key_for_ext_info(symbol: &Symbol) -> Vec<u8> {
+    format!("/config/ext_info/{symbol}").into_bytes()
+}
+
+pub struct SymbolMeta {
```

Comment:
This is the same as `SymbolMetaJson`. Any reason not to use the same type?

Reply:
I want to separate the configuration file layout from our internal structure layout. It is the same right now, but it might not be the case soon™.